mrmustard/lab_dev/states/base.py

```diff
@@ -368,14 +368,16 @@ def quadrature_distribution(self, quad: Vector, phi: float = 0.0) -> ComplexTens
         Returns:
             The quadrature distribution.
         """
-        quad = math.astensor(quad)
+        quad = np.array(quad)
         if len(quad.shape) != 1 and len(quad.shape) != self.n_modes:
             raise ValueError(
                 "The dimensionality of quad should be 1, or match the number of modes."
             )
 
         if len(quad.shape) == 1:
-            quad = math.astensor(list(product(quad, repeat=len(self.modes))))
+            quad = math.astensor(np.meshgrid(*[quad] * len(self.modes))).T.reshape(
```

Comment:
Interesting is it really faster to use meshgrid, transpose and reshape instead? I wouldn't have expected that 😅